server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java

```diff
@@ -262,13 +262,17 @@ public void testCancelFailedSearchWhenPartialResultDisallowed() throws Exception
         // When the search request executes, block all shards except 1.
         final List<SearchShardBlockingPlugin> searchShardBlockingPlugins = initSearchShardBlockingPlugin();
         AtomicBoolean letOneShardProceed = new AtomicBoolean();
```

Comment:
Maybe we could make this slightly simpler by using a n`AtomicReference<CountDownLatch>` and then doing something like  `latch =  getAndSet(null); if (latch != null) latch.await` to save a variable?

Reply:
Not sure I understood how it should work. I think it's better to check if this one works and then if it fixes the issues then we can refactor it.